packages/vitest/src/runtime/run.ts

```diff
@@ -369,7 +369,13 @@ async function runBenchmarkSuite(suite: Suite) {
         }
       })
       benchmark.task!.addEventListener('error', (e) => {
-        defer.reject(e)
+        let causeError
+        const task = e.task
+        const _benchmark = benchmarkMap[task.name || '']
+        if (_benchmark)
+          causeError = task.result!.error
+
+        defer.reject(causeError || e)
```

Comment:
I think this is easier to read with a slightly more declarative style:
```suggestion
        const task = e.task
        const _benchmark = benchmarkMap[task.name || '']
        defer.reject(_benchmark ? task.result!.error : e)
```